File: filter_cluster.py
import pandas as pd
import sys
import numpy as np

data= input("Enter the name of the results folder:")
cluster = input("Enter the name of the cluster:")

n16h = pd.read_csv("/home/chit/Desktop/Thesis/results/{}/cluster_table.csv".format(data))
want = n16h['cluster']=="Cluster {}".format(cluster)

filter = n16h[want]
filter.to_csv("/home/chit/Desktop/Thesis/results/{}/clust{}.csv".format(data,cluster))

print("The cluster is filted and saved as clust{}.csv in {}.".format(cluster, data))

File: filter_cluster.R
suppressMessages(library(AnnotationDbi))
suppressMessages(library(dplyr))
suppressMessages(library(tidyr))
suppressMessages(library(org.Hs.eg.db))

arg <- commandArgs()
result <- arg[6]
input <- arg[7]

clusters = read.csv(sprintf("/nfs/home/students/chit/Thesis/results/%s/cluster_table.csv",result))

filterclust_ensm <- function(clust){
  clu <- clusters %>%
    filter(cluster == clust)
  ens <- AnnotationDbi::select(org.Hs.eg.db,
                               keys = as.character(clu$object),
                               keytype = "SYMBOL",
                               columns = "ENSEMBL")
  return(ens %>% na.omit())
}

whatever <- filterclust_ensm(sprintf("Cluster %s", input))
writeLines(whatever$ENSEMBL, sprintf("/nfs/home/students/chit/Thesis/results/%s/clust%s.txt",result,input))



